jsonrpc/websocket_test.go

```diff
@@ -20,7 +20,7 @@ func testConnection(t *testing.T, ctx context.Context, method jsonrpc.Method, li
 	require.NoError(t, rpc.RegisterMethods(method))
 
 	// Server
-	srv := httptest.NewServer(jsonrpc.NewWebsocket(rpc, utils.NewNopZapLogger()))
+	srv := httptest.NewServer(jsonrpc.NewWebsocket(rpc, nil, utils.NewNopZapLogger()))
```

Comment:
According to [this article](https://www.codingexplorations.com/blog/mastering-nil-channels-in-go-enhance-your-concurrency-control) `nil` channel will block indefinitely, which is I think counter-intuitive.
Maybe it's worth to add a comment that it will block?

Reply:
That's true. The main point here is that in tests we pass nil for arguments in constructor that for anything that is not relevant to a test. 

Nil shutdown channel won't cause any trouble here either because goroutine that is listening for it will be canceled once call `ServeHTTP` is done. Here is more detailed walkthrough:
```
ctx, cancel := context.WithCancel(r.Context())
defer cancel() // 2. context is cancelled when function finished its execution (`ServeHTTP`)
go func() {
		select {
		case <-ws.shutdown: // 1. blocked indefinetely
			cancel()
		case <-ctx.Done(): // 3. since context is canceled on step 2 we exit here, releasing gorutine
		}
}()
```